### PR TITLE
Add user specifc data to token hash to avoid privilige escalation

### PR DESCRIPTION
**Pull request by ceeram**
Add user specifc data to token hash to avoid privilige escalation

Targeted against token-hmac branch of #11101

### CODE REVIEW

**Review by markstory — APPROVED** [review]
Looks good to me.

File: src/View/Helper/SecureFieldTokenTrait.php [hunk 1]
```diff
@@ -55,7 +55,8 @@ protected function _buildFieldToken($url, $fields, $unlockedFields = [])
         $hashParts = [
             $url,
             serialize($fields),
-            $unlocked
+            $unlocked,
+            session_id(),
```
Comment (inoas) [inline src/View/Helper/SecureFieldTokenTrait.php]: Was Security Component requireing native PHP sessions before? If not, could the session id be fetched via https://api.cakephp.org/3.5/class-Cake.Network.Session.html#_id for more flexibility?

Comment (rchavik) [reply src/View/Helper/SecureFieldTokenTrait.php]: it's the same thing:

https://api.cakephp.org/3.5/source-class-Cake.Network.Session.html#456

Comment (ravage84) [reply src/View/Helper/SecureFieldTokenTrait.php]: I would have preferred a wrapped call, too...

**Comment by ravage84** [comment]
Pardon me ignorance, but why was there no changes to the unit tests?

**Comment by markstory** [comment]
Oh there were after I merged this, as the tests got super sad.

**Pull request merged**